src/api/controllers/v1/nls/workflow.go

```diff
@@ -139,3 +140,40 @@ func (u WorkflowController) RerunWorkflow(c *gin.Context) {
 	}
 	c.Status(200)
 }
+
+// GetWorkflowByName
+//	@Summary	Get a workflow by name
+//	@Param		name	path	string	true	"name of workflow"
+//	@Tags		Workflow Management
+//	@Accept		json
+//	@Produce	json
+//	@Success	200	{object}	v1alpha1.Workflow
+//	@Failure	400	{object}	utils.ResponseError
+//	@Failure	404	{object}	utils.ResponseError
+//	@Failure	500	{object}	utils.ResponseError
+//	@Router		/nls/v1/workflows/{name} [get]
+func (u WorkflowController) GetWorkflowByName(c *gin.Context) {
+	wfName := c.Param("name")
+	
+	var workflow interface{}
+	var err error
+	retries := 3
+	delay := 30 * time.Second
+
+	// Retry loop
+	for i := 0; i < retries; i++ {
+		workflow, err = u.service.GetWorkflowByName(wfName, c)
+		if err == nil {
+			c.JSON(200, workflow)
+			return
+		}
```

Comment:
add an else with debug message that attempt failed

Reply:
there are no messages being logged in this whole file, so can we skip printing any message? otherwise I will have to setup the logger.

Reply:
Cool